ledger/src/shred.rs

```diff
@@ -691,7 +691,7 @@ pub fn should_discard_shred(
     };
     let slot = match layout::get_slot(shred) {
         Some(slot) => {
-            if slot <= root || !slot_bounds.contains(&slot) {
```

Comment:
Coding shreds - moved this check to later on once we know what kind of shred we have
Data shreds - this is later checked by calling `verify_shred_slots()`
https://github.com/solana-labs/solana/blob/36d13af015c2830530dd68faae770a558aee0839/ledger/src/blockstore.rs#L4029-L4036

`root <= parent && parent < slot` ==> `root <= parent < slot` ==> `root < slot` ==> `slot > root`. In old logic, we error if `slot <= root` so it must be that `slot > root`